tests/suites/storage/persistent_storage.sh

```diff
@@ -0,0 +1,145 @@
+# This function tests charm deployment with a single filesystem storage unit and a single persistent block device
+# storage unit.
+#  Steps taken to the test:
+#       - Deploy dummy-storage charm with a single block storage unit (ebs)
+#         and a single filesystem storage unit (rootfs).
+#       - Check charm status once the deployment is done.
+#           > Application status should be active.
+#       - Check charm storage units once the deployment is done.
+#           > Total number of storage units should be 2.
+#           > Name of storage units should be in align with charm config.
+#           > Properties of storage units should be as defined.
+#               - Storage Type, Persistent Setting and Pool.
+run_persistent_storage() {
+	echo
+
+	model_name="persistent-storage"
+	file="${TEST_DIR}/test-${model_name}.log"
+	ensure "${model_name}" "${file}"
+	# dummy-storage is going to be deployed with 1 ebs block storage unit
+	# and 1 rootfs filesystem storage unit.
+	echo "dummy-storage is going to be deployed with 1 ebs block storage unit and 1 rootfs filesystem storage unit"
+	juju --debug deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-blk=ebs \
+		--storage single-fs=rootfs
+	# wait for current application-status to be active
+	wait_for "active" '.applications | .[] | ."application-status" | .current'
+	# wait for current workload-status to be active
+	wait_for "active" "$(workload_status "dummy-storage" 0).current"
```

Comment:
You can probably use a helper function from wait-for.sh (i.e. idle_condition) for the query here

Reply:
This will also make the following 2 lines redundant

Reply:
It was a direct translation of this https://github.com/juju/juju/blob/0521fd12a4/acceptancetests/assess_persistent_storage.py#L246-L247. Is there a chance that either might be active while the other is not?

Reply:
So your asserts/the python-based asserts are slightly different to what waiting for an idle condition would do

idle_condition waits for an application's unit to both have juju-status "idle" and also workload-status not error. It's what we tend

In practise it may as well be equivalent, and it's what we use throughout our testing suite to wait for an application to be deployed

Reply:
This would remove a couple of application-status + workload status assertions and join them into one single assertion and still ensure that we are still testing for the correct behaviour. ty for this.